extension/include/boost/di/extension/injections/shared_factory.hpp

```diff
@@ -68,14 +87,14 @@ struct shared_factory_impl {
 #endif
 };
 
-template <class T, class TFunc>
+template <class T, class TPreventRecursion = void, class TFunc>
```

Comment:
Did you consider reverse the logic here and use true_type, false_type, instead?
```
class TPreventRecursion = std::false_type;
```

Reply:
Sure, will change